packages/site/cms-site/src/sitePreview/SitePreviewUtils.ts

```diff
@@ -0,0 +1,63 @@
+import jsonwebtoken from "jsonwebtoken";
+import { cookies, draftMode } from "next/headers";
+import { redirect } from "next/navigation";
+import { type NextRequest } from "next/server";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Scope = Record<string, any>;
+type GraphQLClient = {
+    setHeader(key: string, value: string): unknown;
+    request(document: string, variables?: unknown): Promise<{ isAllowedSitePreview: boolean }>;
+};
+export type SitePreviewData = {
+    includeInvisible: boolean;
+};
+export type SitePreviewParams = {
+    scope: Scope;
+    previewData?: SitePreviewData;
+};
+
+const previewScopeSigningKey = "random"; // TODO improve randomness
```

Comment:
We shouldn't publish stable Comet v7 before changing this

Reply:
yes, it's a point on the agenda for tomorrow